anyway/parsers/twitter.py

```diff
@@ -0,0 +1,114 @@
+    if tweets_df.empty:
+        return tweets_df
+    tweets_df['accident_time'] = tweets_df['tweet_text'].apply(extract_accident_time)
+    tweets_df['accident_date'] = tweets_df['tweet_ts'].apply(lambda ts: datetime.datetime.date(ts))
+
+    tweets_df['link'] = tweets_df['tweet_id'].apply(lambda t: 'https://twitter.com/{}/status/{}'.format(screen_name, t))
+    tweets_df['author'] = [to_hebrew[screen_name]] * len(tweets_df)
+    tweets_df['description'] = [None] * len(tweets_df)  # TODO: Maybe swap description and title
+    tweets_df['source'] = ['twitter'] * len(tweets_df)
+
+    tweets_df['date'] = tweets_df['accident_date'].astype(str) + ' ' + tweets_df['accident_time']
+    tweets_df['location'] = tweets_df['tweet_text'].apply(manual_filter_location_of_text)
+    tweets_df['google_location'] = tweets_df['location'].apply(geocode_extract, args=(google_maps_key,))
+
+    # expanding google maps dict results to separate columns
+    tweets_df = pd.concat([tweets_df, tweets_df['google_location'].apply(pd.Series)], axis=1)
+    tweets_df = pd.concat([tweets_df.drop(['geom'], axis=1), tweets_df['geom'].apply(pd.Series)], axis=1)
+    tweets_df['resolution'] = tweets_df.apply(lambda row: set_accident_resolution(row), axis=1)
+
+    tweets_df.rename(
+        {'tweet_text': 'title', 'lng': 'lon', 'road_no': 'geo_extracted_road_no', 'street': 'geo_extracted_street',
+         'intersection': 'geo_extracted_intersection', 'city': 'geo_extracted_city', 'address': 'geo_extracted_address',
+         'district': 'geo_extracted_district'}, axis=1, inplace=True)
+
+    tweets_df['location_db'] = tweets_df.apply(
+        lambda row: get_db_matching_location(row['lat'], row['lon'], row['resolution'], row['geo_extracted_road_no']),
+        axis=1)
+    tweets_df = pd.concat([tweets_df, tweets_df['location_db'].apply(pd.Series)], axis=1)
+
+    tweets_df.drop(['google_location', 'accident_date', 'accident_time', 'tweet_ts', 'location_db'],
+                   axis=1, inplace=True)
+
+    return tweets_df
+
+
+def scrape_extract_store(screen_name, google_maps_key, db):
+
+    latest_tweet_id = db.get_latest_tweet_id() or 'no_tweets'
+
+    raw_tweets = scrape(screen_name, latest_tweet_id)
+
+    tweets_df = extract_features(raw_tweets, screen_name, google_maps_key)
+    if tweets_df.empty:
+        # NB: why do we avoid inserting non-accidents to the DB?
+        return
+
+    tweets = tweets_df.loc[:, ['tweet_id', 'title', 'link', 'date', 'author', 'description', 'location', 'lat',
+                               'lon', 'resolution', 'region_hebrew', 'district_hebrew', 'yishuv_name',
+                               'street1_hebrew',
+                               'street2_hebrew', 'non_urban_intersection_hebrew', 'road1', 'road2',
+                               'road_segment_name',
+                               'accident', 'source']]
+
+    for row in tweets.itertuples(index=False):
+        db.insert_new_flash_news(**row)
```

Comment:
This is completely untested, and it must be - but I'm not sure how. I don't want to populated the DB with garbage.

Reply:
just a suggestion,
you can monkey patch tweets and set a return value

Reply:
I'm not sure I understand. How does that test the interface with the DB?

Reply:
I don't think we want to test *here* the insertion to the DB, this PR is around re-factoring the twitter scraping, right? you should ensure your calling `insert_new_flash_news` with the correct parameters and assume it's working as expected. (of course some validation should be made on `test/dev/staging` environment before going with this to production.

Reply:
You are right, I agree. There should be a test validating the arguments. Using `NewsFlash` is probably the right way.

Reply:
That would have to be deferred to another PR though, so this PR won't get too big.